types/react-autosuggest/index.d.ts

```diff
@@ -199,16 +187,44 @@ declare namespace Autosuggest {
          * Use it, for example, if you want to display suggestions when input value is at least 2 characters long.
          */
         shouldRenderSuggestions?: ShouldRenderSuggestions;
+        /**
+         * Use your imagination to style the Autosuggest.
+         */
+        theme?: Theme;
+    }
+
+    interface AutosuggestPropsSingleSection<TSuggestion> extends AutosuggestPropsBase<TSuggestion> {
+        /**
+         * Set it to true if you'd like to display suggestions in multiple sections (with optional titles).
+         */
+        multiSection?: false;
         /**
          * These are the suggestions that will be displayed. Items can take an arbitrary shape.
          */
         suggestions: ReadonlyArray<TSuggestion>;
+    }
+
+    interface AutosuggestPropsMultiSection<TSuggestion, TSection> extends AutosuggestPropsBase<TSuggestion> {
         /**
-         * Use your imagination to style the Autosuggest.
+         * Set it to true if you'd like to display suggestions in multiple sections (with optional titles).
          */
-        theme?: Theme;
+        multiSection: true;
+        /**
+         * These are the suggestions that will be displayed. Items can take an arbitrary shape.
+         */
+        suggestions: ReadonlyArray<TSection>;
+        /**
+         * Implement it to teach Autosuggest where to find the suggestions for every section.
+         */
+        getSectionSuggestions?: GetSectionSuggestions<TSuggestion, TSection>;
+        /**
+         * Use your imagination to define how section titles are rendered.
+         */
+        renderSectionTitle?: RenderSectionTitle;
     }
 
+    type AutosuggestProps<TSuggestion, TSection> = AutosuggestPropsSingleSection<TSuggestion> | AutosuggestPropsMultiSection<TSuggestion, TSection>;
```

Comment:
I thought about making this a conditional type:

```ts
type AutosuggestProps<TSuggestion, TSection> = [TSection] extends [never]
  ? AutosuggestPropsSingleSection<TSuggestion>
  : AutosuggestPropsMultiSection<TSuggestion, TSection>;
```

This would preserve outside usage of this type. However, this needs to be a union for the component's props type.